### PR TITLE
delta ak7448

### PR DESCRIPTION
**Pull request by delta-JohnsonLu**
The machine have 48x10GbE SFP28 and 4x100GbE QSFP28 on broadcom BCM56768 Maverick switch ASIC.
CPU module is Intel Rangeley C2338 and with SSD 8G.

### CODE REVIEW

File: packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/sysi.c [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    for (i = 1; i <= NUM_OF_PSU_ON_PSU_BOARD; i++)
+    {
+        *e++ = ONLP_PSU_ID_CREATE(i);
+    }
+
+    return 0;
+}
+
+int
+onlp_sysi_platform_manage_fans(void)
```
Comment (jnealtowns) [inline packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/sysi.c]: The API lock is not held while executing the platform fan and led management functions. 
The fani, thermali, and ledi implementations cannot be called here. Please use the normal fan, thermal, and led calls.

Comment (delta-JohnsonLu) [reply packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/sysi.c]: Hi @jnealtowns:
Thank for your comment but we have some question.
 1. The ONLPI Driver APIs are called by ONLP Application APIs and we see there are API lock function in ONLP 
     Application  APIs. Why do we need to add the lock function in ONLPI Driver APIs?
 2.  We do not know about "normal fan. thermal and led calls" exactly. You means to call the ONLP Application APIs in 
        sysi.c, is it appropriate?

Comment (jnealtowns) [reply packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/sysi.c]: You are correct. All ONLPI implementations are protected against concurrency by the upper ONLP layer through shared mutex locks. 

The *only* ones which *aren't* protected are the onlp_sysi_platform_manage_* functions. They are poorly named to reflect this.

Comment (delta-JohnsonLu) [reply packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/sysi.c]: I will update the onlp_sysi_platform_manage_* functions which reflect the protection while calling ONLPI.

**Comment by delta-JohnsonLu** [comment]
New commits update the onlp_sysi_platform_manage_* functions which reflect the protection while calling ONLPI.

File: packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/sysi.c [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+{
+    int i = 0;
+    int new_percentage;
+    int highest_temp = 0;
+    onlp_thermal_info_t thermal[NUM_OF_THERMAL_ON_BOARDS];
+
+    /* Get current temperature */
+    if (onlp_thermali_info_get(ONLP_THERMAL_ID_CREATE(THERMAL_CPU_CORE),        &thermal[0]) != ONLP_STATUS_OK ||
```
Comment (delta-JohnsonLu) [inline packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/sysi.c]: The onlp_thermali_info_get() in onlp_sysi_platform_manage_*  read  the attribute by  dni_i2c_lock_read_attribute() function.
https://github.com/delta-JohnsonLu/OpenNetworkLinux/blob/b92a1c57e114f2a6e321bddaf370ce5fc60d7eed/packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/platform_lib.c#L130

File: packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/thermali.c [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    }
+
+    mux_info.bus = I2C_BUS_5;
+    mux_info.addr = CPLD_B;
+    mux_info.offset = offset;
+    mux_info.channel = channel;
+    mux_info.flags = DEFAULT_FLAG;
+    
+    sprintf(fullpath, "%s%s", PREFIX_PATH, last_path[local_id]);
+    r_data = dni_i2c_lock_read_attribute(&mux_info, fullpath);
```
Comment (delta-JohnsonLu) [inline packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/thermali.c]: Call onlp_thermali_info_get() to read  the attribute by dni_i2c_lock_read_attribute() function.

File: packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/platform_lib.c [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+    pthread_mutex_unlock(&mutex);
+    return 0;
+}
+
+int dni_i2c_lock_read_attribute(mux_info_t * mux_info, char * fullpath)
+{
+    int fd, len, nbytes = 10;
+    char  r_data[10]   = {0};
+
+    pthread_mutex_lock(&mutex);
```
Comment (delta-JohnsonLu) [inline packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/platform_lib.c]: When call dni_i2c_lock_read_attribute() shares the same mutex lock.

**Pull request merged**